src/dynamixel/servos/servo.hpp

```diff
@@ -195,7 +195,7 @@ namespace dynamixel {
             {
                 typename Servo<Model>::ct_t::present_position_t pos;
                 Servo<Model>::protocol_t::unpack_data(st.parameters(), pos);
-                double deg = ((pos - ct_t::min_goal_position) * (ct_t::max_goal_angle_deg - ct_t::min_goal_angle_deg) / (ct_t::max_goal_position - ct_t::min_goal_position)) + ct_t::min_goal_angle_deg;
+                double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg;
```

Comment:
You should use `static_cast` here.

Reply:
seems like a bit overkill but why not? better be safe

Reply:
will

from
```
double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg; 
```
to

```
double deg = ((static_cast<double>(pos) - static_cast<double>(ct_t::min_goal_position)) * (static_cast<double>(ct_t::max_goal_angle_deg) - static_cast<double>(ct_t::min_goal_angle_deg)) / (static_cast<double>(ct_t::max_goal_position) - static_cast<double>(ct_t::min_goal_position))) + static_cast<double>(ct_t::min_goal_angle_deg); 
```
suffice?

Reply:
https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting

Reply:
> will from
> 
> ```
> double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg; 
> ```
> 
> to
> 
> ```
> double deg = ((static_cast<double>(pos) - static_cast<double>(ct_t::min_goal_position)) * (static_cast<double>(ct_t::max_goal_angle_deg) - static_cast<double>(ct_t::min_goal_angle_deg)) / (static_cast<double>(ct_t::max_goal_position) - static_cast<double>(ct_t::min_goal_position))) + static_cast<double>(ct_t::min_goal_angle_deg); 
> ```
> 
> suffice?
> if so I will open another PR to apply this change

Yes, that's right. It is much safer and better implementation in C++.

Reply:
got it will open a PR